NEL.MESH/Services/Orchestrations/Mesh/MeshOrchestrationService.cs

```diff
@@ -72,7 +72,13 @@ public MeshOrchestrationService(ITokenService tokenService, IMeshService meshSer
             });
 
         public ValueTask<List<string>> RetrieveMessagesAsync() =>
-            throw new System.NotImplementedException();
+            TryCatch(async () =>
+            {
+                string token = await this.tokenService.GenerateTokenAsync();
+                List<string> outputMessage = await this.meshService.RetrieveMessagesAsync(authorizationToken: token);
```

Comment:
You are missing the validation.  Add `ValidateToken(token);` like above and add a unit test for it

Reply:
good spot